crates/verify/src/etherscan/flatten.rs

```diff
@@ -165,35 +169,47 @@ Diagnostics: {diags}",
     ///
     /// # Exits
     ///
-    /// If the solc compiler output contains errors, this could either be due to a bug in the
+    /// If the zksolc compiler output contains errors, this could either be due to a bug in the
     /// flattening code or could to conflict in the flattened code, for example if there are
     /// multiple interfaces with the same name.
     fn zk_check_flattened(
         &self,
         content: impl Into<String>,
-        version: &Version,
+        compiler_version: &ZkVersion,
         contract_path: &Path,
     ) -> Result<()> {
-        let version = strip_build_meta(version.clone());
-        let zksolc = ZkSolc::find_installed_version(&version)?
-            .unwrap_or(ZkSolc::blocking_install(&version)?);
+        let solc_version = strip_build_meta(compiler_version.solc.clone());
+        let zksolc_version = strip_build_meta(compiler_version.zksolc.clone());
+        let zksolc = ZkSolc::find_installed_version(&zksolc_version)?
+            .unwrap_or(ZkSolc::blocking_install(&solc_version)?);
 
         let input = ZkSolcVersionedInput {
             input: ZkSolcInput {
                 language: SolcLanguage::Solidity,
                 sources: Sources::from([("contract.sol".into(), Source::new(content))]),
                 ..Default::default()
             },
-            solc_version: version.clone(),
+            solc_version: solc_version.clone(),
             allow_paths: Default::default(),
             base_path: Default::default(),
             include_paths: Default::default(),
         };
 
-        let out = zksolc.compile(&input.input)?;
+        let solc_compiler = if compiler_version.is_zksync_solc {
+            // AutoDetect given a specific solc version on the input, will
+            // find or install the solc version
+            SolcCompiler::AutoDetect
```

Comment:
In this case it will autodetect the zksync's fork of solc to use, right?

Reply:
yes because the solc version is being passed in the input